src/socket/address.cr

```diff
@@ -364,25 +401,26 @@ class Socket
       0 <= field <= 0xff ? field.to_u8! : raise Error.new("Invalid IPv4 field: #{field}")
     end
 
-    # Returns the IPv6 address with the given address *fields* and *port*
-    # number.
-    def self.v6(fields : UInt16[8], port : UInt16) : self
+    # Returns the IPv6 address with the given address *fields*, *port* number
+    # and scope identifier.
+    def self.v6(fields : UInt16[8], port : UInt16, zone_id : UInt32 = 0u32) : self
```

Comment:
```suggestion
    def self.v6(fields : UInt16[8], port : UInt16, zone_id : Int = 0) : self
```

Reply:
Depends on type discussion/question in https://github.com/crystal-lang/crystal/pull/15263#issuecomment-2887607827